flexget/plugins/metainfo/nfo_lookup.py

```diff
@@ -0,0 +1,241 @@
+        if not config:
+            # Config was set to 'no' instead of yes. Don't do anything then.
+            return
+
+        for entry in task.entries:
+            # If this entry was obtained from the filesystem plugin it should
+            # have a filename field. If it does not have one then there is
+            # nothing we can do in this plugin.
+            filename = entry.get('filename')
+            location = entry.get('location')
+
+            # If there is no 'filename' field there is also no nfo file
+            if filename is None or location is None:
+                log.warning("Entry '{0}' didn't come from the filesystem plugin".format(entry.get('title')))
+                nfo_filename = None
+            else:
+                nfo_filename = self.get_nfo_filename(entry)
+                # Stop if there is no nfo file
+                if nfo_filename is None:
+                    log.warning(
+                        "Entry '{0}' has no corresponding '.nfo' file".format(
+                            entry.get('title')))
+                    # continue
+
+            # Populate the fields from the information in the .nfo file
+            self.lookup(entry, nfo_filename)
+
+    def lookup(self, entry, nfo_filename):
+        # If there is already data from a previous parse then we don't need to
+        # do anything
+        if entry.get('nfo_id') is not None:
+            log.warning("Entry '{0}' was already parsed by nfo_lookup. "
+                        "I won't do anything.".format(entry.get('title')))
+            return
+
+        # In case there is no nfo file, `nfo_filename` will be None at this point
+        if nfo_filename is None:
+            # Setting the fields dictionary to be an empty dictionary here
+            # means that no new information will be added to the entry in this
+            # lookup function. We will still perform an IMDB lookup later, but
+            # it will be the same as if we had used the imdb_lookup plugin.
+            fields = {}
+        else:
+            # Get all values we can from the nfo file
+            fields = self.get_fields_from_nfo_file(nfo_filename)
+
+            # Update the entry with the just the imdb_id. This is enouth so that we
+            # can search with ImdbLookup and be sure to get the same movie pointed
+            # by the nfo file
+            if 'nfo_id' in fields:
+                entry.update({u'imdb_id': fields['nfo_id']})
+            else:
+                # We will perform an IMDB search later the fields add here might be
+                # overwritten
+                self.add_imdb_fields_to_dict(fields)
+
+        # Let's search metadata in IMDB. If we found an nfo file containing an
+        # ID we have already added the 'imdb_id' field and this will make the
+        # IMDB lookup below find the correct movie no matter what the title and
+        # filename are.
+        imdb_lookup = ImdbLookup()
```

Comment:
yeah, if `imdb_id` is populated it will be used for lookup by `imdb_lookup`. to make it run before lookups look into the `@priority` decorator